core/plugins/org.polarsys.capella.core.data.migration/src/org/polarsys/capella/core/data/migration/aird/AssociationCDBMigrationContributor.java

```diff
@@ -0,0 +1,368 @@
+    View sourceView = gmfEdge.getSource();
+    View targetView = gmfEdge.getTarget();
+    List<Edge> sourceViewSourceEdges = sourceView.getSourceEdges();
+    List<Edge> sourceViewTargetEdges = sourceView.getTargetEdges();
+    List<Edge> targetViewSourceEdges = targetView.getSourceEdges();
+    List<Edge> targetViewTargetEdges = targetView.getTargetEdges();
+    Anchor sourceAnchor = gmfEdge.getSourceAnchor();
+    Anchor targetAnchor = gmfEdge.getTargetAnchor();
+    RelativeBendpoints bendpoints = (RelativeBendpoints) gmfEdge.getBendpoints();
+    List<?> bendpointsList = bendpoints.getPoints();
+    List<?> revBendpointsList = getReversedBendpoints(bendpointsList);
+
+    // labels
+    List<?> edgeChildren = gmfEdge.getChildren();
+    Optional<Node> beginLabel = edgeChildren.stream().filter(this::isBeginLabel).map(Node.class::cast).findAny();
+    Optional<Node> middleLabel = edgeChildren.stream().filter(this::isMiddleLabel).map(Node.class::cast).findAny();
+    Optional<Node> endLabel = edgeChildren.stream().filter(this::isEndLabel).map(Node.class::cast).findAny();
+
+    // changes
+    beginLabel.ifPresent(label -> migrateLabelPosition(label, bendpointsList, revBendpointsList));
+    middleLabel.ifPresent(label -> migrateLabelPosition(label, bendpointsList, revBendpointsList));
+    endLabel.ifPresent(label -> migrateLabelPosition(label, bendpointsList, revBendpointsList));
+
+    beginLabel.ifPresent(label -> label.setType(SiriusVisualIDRegistry.getType(DEdgeEndNameEditPart.VISUAL_ID)));
+    endLabel.ifPresent(label -> label.setType(SiriusVisualIDRegistry.getType(DEdgeBeginNameEditPart.VISUAL_ID)));
+
+    gmfEdge.setSource(targetView);
+    gmfEdge.setTarget(sourceView);
+    sourceViewTargetEdges.add(gmfEdge);
+    targetViewSourceEdges.add(gmfEdge);
+    sourceViewSourceEdges.remove(gmfEdge);
+    targetViewTargetEdges.remove(gmfEdge);
+
+    gmfEdge.setSourceAnchor(targetAnchor);
+    gmfEdge.setTargetAnchor(sourceAnchor);
+    bendpoints.setPoints(revBendpointsList);
+  }
+  
+  private void migrateAssociation(DEdge dEdge) {
+    migrateDEdge(dEdge);
+    migrateGMFEdge(SiriusGMFHelper.getGmfEdge(dEdge));
+  }
+
+  private boolean needMigration(DEdge dEdge) {
+    if (dEdge.getTarget() instanceof Association association) {
+      Property sourceProp = InformationServices.getService().getAssociationSource(association);
+      Property targetProp = InformationServices.getService().getAssociationTarget(association);
+      AbstractType expectedSource = sourceProp.getAbstractType();
+      AbstractType expectedTarget = targetProp.getAbstractType();
+      if (dEdge.getSourceNode() instanceof DSemanticDecorator sourceNode && dEdge.getTargetNode() instanceof DSemanticDecorator targetNode) {
+        EObject sourceObject = sourceNode.getTarget();
+        EObject targetObject = targetNode.getTarget();
+        return sourceObject == expectedTarget && targetObject == expectedSource;
+      } else {
```

Comment:
A simple "return false;" for all other cases is enough.

Reply:
Proposition:
```
if (dEdge.getTarget() instanceof Association association) {
      ...
      if (dEdge.getSourceNode() instanceof DSemanticDecorator sourceNode && dEdge.getTargetNode() instanceof DSemanticDecorator targetNode) {
        ...
        return sourceObject == expectedTarget && targetObject == expectedSource;
      }
}
// not association or wrong conditions
return false;
```

Reply:
I'm not sure it's easier to read (at least not for me). This "return" fallback principle gives the impression at first sight that `false` is always returned, and confuses the cases where `false` is actually returned. For example, it gets harder to read what's going on if `dEdge.getTarget() instanceof Association association` is `true` and `dEdge.getSourceNode() instanceof DSemanticDecorator sourceNode && dEdge.getTargetNode() instanceof DSemanticDecorator targetNode` is `false`.

Reply:
With a "more strict" Checkstyle configuration, like in Sirius for example, you can not have three "return" statements.

Reply:
But OK with your version (authorized in Capella code style).